app/book.rb

```diff
@@ -1,15 +1,15 @@
 require_relative '.\item'
 
 class Book < Item
-  attr_accessor :cover_state
+  attr_accessor :name, :publish_date, :cover_state, :publisher
```

Comment:
Please consider changing `publish_date` to `published_date`, since it is not preserving the published date data while adding a new book.

Reply:
I have fixed the error